### PR TITLE
[Projects] Refactor cluster specification

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

This introduces a `config` key in the cluster specification to make room for an additional `params` field that can be used to configure clusters.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by mitchellstern — COMMENTED** [review]
Can you remove the quotes from all instances of `config: "cluster.yaml"` for consistency with the rest of the YAML file?

File: python/ray/projects/projects.py [hunk 1]
```diff
@@ -50,6 +50,44 @@ def working_directory(self):
         directory = os.path.join("~", self.config["name"], "")
         return directory
 
+    def make_argument_parser(self, name, params, wildcards):
```
Comment (mitchellstern) [inline python/ray/projects/projects.py]: It doesn't look like `self` is getting used in this function, should we move it outside of the class definition?

File: python/ray/projects/projects.py [hunk 2]
```diff
@@ -175,10 +192,12 @@ def check_project_config(project_root, project_config):
 
     # Make sure the cluster yaml file exists
     if "cluster" in project_config:
-        cluster_file = os.path.join(project_root, project_config["cluster"])
-        if not os.path.exists(cluster_file):
-            raise ValueError("'cluster' file does not exist "
-                             "in {}".format(project_root))
+        if "config" in project_config["cluster"]:
```
Comment (mitchellstern) [inline python/ray/projects/projects.py]: If we've already validated the `project_config` against the JSON schema, then `project_config["cluster"]["config"] should be guaranteed to exist and we can skip these checks, right?

File: python/ray/projects/projects.py (hunk 1 above)
Comment (pcmoritz) [reply python/ray/projects/projects.py]: done

File: python/ray/projects/projects.py (hunk 2 above)
Comment (pcmoritz) [reply python/ray/projects/projects.py]: Yeah, that's right! I did it :)

**Review by mitchellstern — APPROVED** [review]

**Pull request merged**